chainstate/storage/src/mock/mod.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A mock version of the blockchain storage.
+
+mod mock_impl;
+mod mock_impl_accounting;
+
+pub use mock_impl::{MockStore, MockStoreTxRo, MockStoreTxRw};
+
+#[cfg(test)]
+mod tests {
```

Comment:
These tests could be moved into a separate file to keep the top-level module focused on exposing external interfaces